client/src/main/java/io/pravega/client/control/impl/ControllerImpl.java

```diff
@@ -1508,6 +1513,31 @@ private TxnSegments convert(CreateTxnResponse response) {
                 });
     }
 
+    @Override
+    public CompletableFuture<List<TransactionInfo>> listCompletedTransactions(final Stream stream) {
+        Exceptions.checkNotClosed(closed.get(), this);
+        Preconditions.checkNotNull(stream, "stream");
+        final long requestId = requestIdGenerator.get();
+        long traceId = LoggerHelpers.traceEnter(log, LIST_COMPLETED_TRANSACTIONS, stream, requestId);
+
+        try {
+            final CompletableFuture<ListCompletedTxnResponse> listCompletedTxnResponse = this.retryConfig.runAsync(() -> {
```

Comment:
is there any possibility of exception here thrown by the calling method? how are we handling that?

Reply:
There is no possibility to get an exception. If any exception occurs while executing the api, It will get handled in ControllerServiceImpl class (while converting response to gRPC's Observer pattern).